### PR TITLE
Emit freshnessLag metric from broker for queries hitting consuming segments

### PR DESCRIPTION
**Pull request by sunithabeeram**
This is the next (mostly last) phase of the implementation phase for freshness metric discussed in #4007. 

This change ties in the previous changes #4025, #3979 to emit a metric from the broker. See issue and corresponding [design doc](https://cwiki.apache.org/confluence/display/PINOT/Pinot+Freshness+Metric) for more details.

A change from #3979 is to use -1 as the invalid value instead of long MIN/MAX values to keep logging simpler.

Testing done:
* Instrumented internal kafka stream implementation to include ingestion time in RowMetadata.
* Verified the logging for cases where there was no delay; also manually induced delay in consumption across segments and verified that the logging and metric reported the right values.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java [hunk 1]
```diff
@@ -182,7 +187,9 @@ public BrokerResponseNative reduceOnDataTable(@Nonnull BrokerRequest brokerReque
     brokerResponseNative.setNumGroupsLimitReached(numGroupsLimitReached);
     if (numConsumingSegmentsQueried > 0) {
       if (minConsumingFreshnessTimeMs == Long.MAX_VALUE) {
-        LOGGER.error("Invalid lastIndexedTimestamp across {} consuming segments", numConsumingSegmentsQueried);
+        LOGGER.error("Invalid freshness time across {} consuming segments", numConsumingSegmentsQueried);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java]: If numConsumingSegmentsQueried > 0, minConsumingFreshnessTimeMs will never be invalid.

File: pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java [hunk 2]
```diff
@@ -170,11 +170,13 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
       }
     }
 
-    long minConsumingFreshnessTimeMs = Long.MAX_VALUE;
+    long minConsumingFreshnessTimeMs = -1;
```
[the remark this answers is not in the document]
Comment (sunithabeeram) [reply pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java]: I thought I was logging it regardless, but it does go through the dataTable between the ServerV1Impl and QueryScheduler where it is logged. Will remove it.

File: pinot-common/src/main/java/org/apache/pinot/common/response/broker/BrokerResponseNative.java [hunk 3]
```diff
@@ -208,6 +208,7 @@ public void setNumSegmentsMatched(long numSegmentsMatched) {
   }
 
   @JsonProperty("numConsumingSegmentsQueried")
+  @Override
   public long getNumConsumingSegmentsQueried() {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/response/broker/BrokerResponseNative.java]: Not related to this PR, but can you add override this to getNumSegmentsQueried, getNumSegmentsProcessed, getNumSegmentsMatched as well?

File: pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QueryScheduler.java [hunk 4]
```diff
@@ -192,10 +197,11 @@ public void stop() {
 
     if (queryLogRateLimiter.tryAcquire() || forceLog(schedulerWaitMs, numDocsScanned)) {
       LOGGER.info(
-          "Processed requestId={},table={},segments(queried/processed/matched)={}/{}/{},schedulerWaitMs={},totalExecMs={},totalTimeMs={},broker={},numDocsScanned={},scanInFilter={},scanPostFilter={},sched={}",
-          requestId, tableNameWithType, numSegmentsQueried, numSegmentsProcessed, numSegmentsMatched, schedulerWaitMs,
+          "Processed requestId={},table={},segments(queried/processed/matched/consuming)={}/{}/{}/{},schedulerWaitMs={},totalExecMs={},totalTimeMs={},minConsumingFreshnessMs={},broker={},numDocsScanned={},scanInFilter={},scanPostFilter={},sched={}",
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QueryScheduler.java]: Can you re-format this statement?

**Pull request merged**